### PR TITLE
Updated pom to import changes from private-api-sdk-java.

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

Updated pom to import changes from private-api-sdk-java.

Ported over the new and improved exception handling logic that was introduced in accounts-user-api.

Deleted redundant code.

Will need to reinstate ControllerAdvice tests, after the first endpoint has been built.

__JIRA Ticket Number__

IDVA6-599

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [x] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/accounts/association/models/Association.java
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/accounts/association/models/Association.java:1]: we still need this.

File: src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java:1]: need this.

File: src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java:1]: need this

File: src/test/java/uk/gov/companieshouse/accounts/association/controllers/ControllerAdviceTest.java
Comment (kkonuganti-ch) [inline src/test/java/uk/gov/companieshouse/accounts/association/controllers/ControllerAdviceTest.java:1]: need this

File: src/main/java/uk/gov/companieshouse/accounts/association/models/Association.java
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/accounts/association/models/Association.java:1]: Reinstated

File: src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java:1]: Reinstated

File: src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/accounts/association/service/AssociationsService.java:1]: Reinstated

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by kkonuganti-ch — APPROVED** [review]

**Pull request merged**